src/parser/transform/expression/transform_param_ref.cpp

```diff
@@ -10,9 +10,32 @@ unique_ptr<ParsedExpression> Transformer::TransformParamRef(duckdb_libpgquery::P
 	if (node->number < 0) {
 		throw ParserException("Parameter numbers cannot be negative");
 	}
+
+	if (node->name) {
+		// This is a named parameter, try to find an entry for it
+		D_ASSERT(node->number == 0);
+		int32_t index;
+		if (GetNamedParam(node->name, index)) {
+			// We've seen this named parameter before and assigned it an index!
+			node->number = index;
+		}
+	}
 	if (node->number == 0) {
 		expr->parameter_nr = ParamCount() + 1;
+		if (node->name && !HasNamedParameters() && ParamCount() != 0) {
```

Comment:
Could we add some sqllogictest tests using named parameters with `PREPARE` and `EXECUTE` (it is fine if they fail - mostly just verifying that it doesn't blow up).

Reply:
I've added error detection for this to TransformPrepare, and added a test that shows PREPARE throws when a named parameter is found

Do you think that suffices or does this need more coverage?

Reply:
I think that's fine